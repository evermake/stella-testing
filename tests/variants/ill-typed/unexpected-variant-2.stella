language core;
extend with #lists;
extend with #variants;

fn main(x : Nat) -> Nat {
    return (if <| value = n |> then fn(x : Nat) { return 0 } else fn(x : Nat) { return 0 } )(0)
}