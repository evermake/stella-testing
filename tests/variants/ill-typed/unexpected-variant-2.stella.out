An error occurred during typechecking!
===================
Primary type error:
===================
Type Error Tag: [ERROR_UNEXPECTED_VARIANT]
expected an expression of a non-variant type
  Bool
but got a variant
  <| value = n |>
========================
Alternative type errors:
========================
Type Error Tag: [ERROR_AMBIGUOUS_VARIANT_TYPE]
type inference for variants is not supported (use type ascriptions or enable #structural-subtyping)